### PR TITLE
Initial test setup

### PR DESCRIPTION
**Pull request by priyanshus**
An attempt to add tests using Cypress, WebDriver, Testcafe and Taiko in order to incorporate with https://github.com/getgauge/taiko/issues/501. 

As of now, the code changes contain basic test framework setup and dropdown action test using mentioned automation tools. All the tests can be run by a single command `npm test`. The test report shows the time taken to run every test which can be used to compare the result with Taiko.

Test Execution Report:
Screenshot 2019-04-23 at 3 50 55 PM

### CODE REVIEW

**Comment by priyanshus** [comment]
Let me know your views on the above tests so that I can proceed with other tests.

File: tests/cypress-test/plugins/index.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// ***********************************************************
```
Comment (sriv) [inline tests/cypress-test/plugins/index.js]: This file looks like a boilerplate generated. It has no working code, perhaps this is worth cleaning up?

File: tests/cypress-test/support/commands.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// ***********************************************
```
Comment (sriv) [inline tests/cypress-test/support/commands.js]: This file looks like a boilerplate generated. It has no working code, perhaps this is worth cleaning up?

File: tests/cypress-test/support/index.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+// ***********************************************************
```
Comment (sriv) [inline tests/cypress-test/support/index.js]: This file looks like a boilerplate generated. It has no working code, perhaps this is worth cleaning up?

**Comment by priyanshus** [comment]
@sriv Updated as per change request.

**Comment by sriv** [comment]
Some further observations:

- can we rename `tests` to `benchmarks` and have a section in the repo's readme on how to run/interpret the benchmarks?

- when I check out this fork and do `cd tests && npm i && npm test` I get this error:

```
npm i && npm test
audited 4002 packages in 2.761s
found 0 vulnerabilities


> webdriver-test@0.0.1 test /tmp/the-internet-express/tests
> mocha ./webdriver-test/specs/**/*.spec.js && ./node_modules/.bin/cypress run --browser chrome && mocha ./taiko-test/specs/**/*.spec.js && ./node_modules/.bin/testcafe chrome ./testcafe-test/specs/**/*.spec.js



  Actions by webdriver
    1) "before all" hook for ".select() - select an option in a <select> element"
    2) "after all" hook for ".select() - select an option in a <select> element"


  0 passing (105ms)
  2 failing

  1) Actions by webdriver
       "before all" hook for ".select() - select an option in a <select> element":
     WebDriverError: unknown error: cannot find Chrome binary
  (Driver info: chromedriver=73.0.3683.20 (8e2b610813e167eee3619ac4ce6e42e3ec622017),platform=Linux 4.18.0-17-generic x86_64)
      at Object.checkLegacyResponse (node_modules/selenium-webdriver/lib/error.js:585:15)
      at parseHttpResponse (node_modules/selenium-webdriver/lib/http.js:533:13)
      at Executor.execute (node_modules/selenium-webdriver/lib/http.js:468:26)
      at processTicksAndRejections (internal/process/task_queues.js:86:5)

  2) Actions by webdriver
       "after all" hook for ".select() - select an option in a <select> element":
     TypeError: Cannot read property 'quit' of undefined
      at Context.after (webdriver-test/specs/actions/actions.spec.js:27:18)
      at processImmediate (internal/timers.js:443:21)



npm ERR! Test failed.  See above for more details.
```

What am I missing?

**Comment by priyanshus** [comment]
> Some further observations:
> 
> * can we rename `tests` to `benchmarks` and have a section in the repo's readme on how to run/interpret the benchmarks?
> * when I check out this fork and do `cd tests && npm i && npm test` I get this error:
> 
> ```
> npm i && npm test
> audited 4002 packages in 2.761s
> found 0 vulnerabilities
> 
> 
> > webdriver-test@0.0.1 test /tmp/the-internet-express/tests
> > mocha ./webdriver-test/specs/**/*.spec.js && ./node_modules/.bin/cypress run --browser chrome && mocha ./taiko-test/specs/**/*.spec.js && ./node_modules/.bin/testcafe chrome ./testcafe-test/specs/**/*.spec.js
> 
> 
> 
>   Actions by webdriver
>     1) "before all" hook for ".select() - select an option in a <select> element"
>     2) "after all" hook for ".select() - select an option in a <select> element"
> 
> 
>   0 passing (105ms)
>   2 failing
> 
>   1) Actions by webdriver
>        "before all" hook for ".select() - select an option in a <select> element":
>      WebDriverError: unknown error: cannot find Chrome binary
>   (Driver info: chromedriver=73.0.3683.20 (8e2b610813e167eee3619ac4ce6e42e3ec622017),platform=Linux 4.18.0-17-generic x86_64)
>       at Object.checkLegacyResponse (node_modules/selenium-webdriver/lib/error.js:585:15)
>       at parseHttpResponse (node_modules/selenium-webdriver/lib/http.js:533:13)
>       at Executor.execute (node_modules/selenium-webdriver/lib/http.js:468:26)
>       at processTicksAndRejections (internal/process/task_queues.js:86:5)
> 
>   2) Actions by webdriver
>        "after all" hook for ".select() - select an option in a <select> element":
>      TypeError: Cannot read property 'quit' of undefined
>       at Context.after (webdriver-test/specs/actions/actions.spec.js:27:18)
>       at processImmediate (internal/timers.js:443:21)
> 
> 
> 
> npm ERR! Test failed.  See above for more details.
> ```
> What am I missing?

I believe you do not have Chrome binaries on your system. 

Check this: `WebDriverError: unknown error: cannot find Chrome binary`

Or, maybe you are having chrome installed at some other location than the default location.

**Comment by sriv** [comment]
Yes, I don't have chrome installed. I think we should not assume that, and use something like `node-chromium` to include the chromium binary as part of `npm install`.

Note that taiko bootstraps the chromium binary as part of it's install. Tests could either reuse the binary setup by taiko or taiko could use a custom chromium binary set by `TAIKO_BROWSER_PATH`

**Comment by priyanshus** [comment]
WebDriver has not started supporting the latest version of chromium yet. Taiko installs the latest version of chromium so we can't use it for webdriver tests.

There is no package available in npm which installs the chromium version of your choice. So either I have to write custom logic to download the older version of chromium or temporarily we can use chrome for webdriver tests. What do you say?

**Comment by sriv** [comment]
I see that there is an open issue for ability to download an older version of chromim. https://github.com/dtolstyi/node-chromium/issues/1

I suppose we can mandate `chrome`/`chromium` as a pre-requisite and make the tests use it. In that case, please add that to the docs.

Also, what do you think of my other comment:

> can we rename tests to benchmarks and have a section in the repo's readme on how to run/interpret the benchmarks?

**Pull request merged**